[tanteckfang] iP

tanteckfang opened the pull request:
### **ChatBot Class**

- easy to use
- able to add task

### Available Task type
1. Todos
2. Deadlines
3. Events

- [X] free to use  😆

> Its always good to be free

Downloaded java from here[https://www.java.com/en/](url)

_Here's a Java method to print "Hello, World!"_: System.out.println("Hello, world!");` 



Here's the main method

>    
```ruby
public static void main(String[] args){
        Chatbot Sara = new Chatbot();
        Sara.run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Complete Level-0

Commit:
done level 1

tanteckfang committed:
Merge pull request #1 from tanteckfang/Level_1

done level 1

Commit:
done level-2

tanteckfang committed:
Merge pull request #2 from tanteckfang/Level_2

done level-2

Commit:
done level-3

Commit:
done level-4

tanteckfang committed:
Merge pull request #3 from tanteckfang/Level_4

Level 4

Commit:
UI testing added

Commit:
done Level 5

Commit:
done level6'

/
>?
q
/q
wq
cdc
'

Commit:
added enum feature

tanteckfang committed:
Merge pull request #4 from tanteckfang/enums

Enums

Commit:
done level 7

Commit:
done level 8

Commit:
done OOP

Commit:
done OOP

Commit:
done OOP

Commit:
done packages

tanteckfang committed:
Merge pull request #5 from tanteckfang/branch-A-pacakages

Branch a pacakages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
add Gradle

Commit:
done JUnit

tanteckfang committed:
Merge pull request #6 from tanteckfang/branch-A-JUnit

Branch a j unit

tanteckfang committed:
done adding Javadoc

tanteckfang committed:
added coding Standard

tanteckfang committed:
done Level-9

tanteckfang committed:
Merge branch 'branch-A-JavaDoc'

tanteckfang committed:
Merge branch 'branch-A-CodingStandard'

tanteckfang committed:
done resolving conflicts

tanteckfang committed:
done level 10

sp4ce-cowboy commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,142 @@
+    /**
+     * Enumeration representing different types of tasks and commands.
+     */
+    public enum TaskType {
```
I like your use of Enums here, makes the code a lot more readable!

sp4ce-cowboy commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,142 @@
+     * @return Task type of the command
+     */
+    public static TaskType determineTaskType(String userInput) {
+        if (userInput.startsWith("todo")) return TaskType.TODO;
```
Perhaps you could consider using egyptian braces as per the coding standards?

sp4ce-cowboy commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,142 @@
+     * @return array of string after parsing the command
+     * @throws ChatbotException if the description or date is empty
+     */
+    public static String[] parseDeadline(String userInput) throws ChatbotException {
```
I like how you have accounted for specific types of errors within each task parser function!

sp4ce-cowboy commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,142 @@
+        TaskType taskType = determineTaskType(userInput);
+        Command command;
+
+        switch (taskType) {
```
Maybe you could consider using same level indentations for the switch and case statements?

sp4ce-cowboy commented on src/main/java/chatbot/Ui.java:
```diff
@@ -0,0 +1,58 @@
+     * print greeting message.
+     */
+    public void showGreeting() {
+        System.out.println("    ____________________________________________________________");
```
Do you think it would be better if the System.out.println(<line>) was abstracted into a "linePrinter" function, since it is called a couple of times?

sp4ce-cowboy commented on src/main/java/chatbot/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,35 @@
+package chatbot.command;
```
I like how you have used sub-packaging in your classes

sp4ce-cowboy commented on src/main/java/chatbot/command/Command.java:
```diff
@@ -0,0 +1,23 @@
+/**
+ * Abstract base class for different types of commands that can be executed by the chatbot.
+ */
+public abstract class Command {
```
Your use of an inheritable Abstract Command class to facilitate polymorphic behaviour and substitution is fantastic!

sp4ce-cowboy commented on src/main/java/chatbot/storage/Storage.java:
```diff
@@ -0,0 +1,87 @@
+                    tasks.add(task);
+                } else {
+                    // Handle corrupted line here for stretch goal
+                    System.out.println("Warning: Corrupted line skipped in file");
```
Your handling of file-related errors is great, I like how you have dealt with Exceptions directly instead of throwing them back to the calling function.

sp4ce-cowboy commented on src/main/java/chatbot/task/Deadlines.java:
```diff
@@ -0,0 +1,50 @@
+/**
+ * class Deadline extends class Task which consist of variable for Deadline.
+ */
+public class Deadlines extends Task {
```
LGTM

sp4ce-cowboy commented on src/main/java/chatbot/task/Events.java:
```diff
@@ -0,0 +1,40 @@
+/**
+ * class Event extends class Task which consist of variable for Event.
+ */
+public class Events extends Task {
```
LGTM

sp4ce-cowboy commented on src/main/java/chatbot/task/Task.java:
```diff
@@ -0,0 +1,116 @@
+     * @param line the line of task
+     * @return task in Task format
+     */
+    public static Task fromFileFormat(String line) {
```
Perhaps you could consider abstracting this static function into another Parser class?

sp4ce-cowboy commented on src/main/java/chatbot/task/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+/**
+ * List of task.
+ */
+public class TaskList {
```
LGTM!

sp4ce-cowboy commented on src/main/java/chatbot/task/TaskManager.java:
```diff
@@ -0,0 +1,216 @@
+        Task task = new Todos(t);
+        tasks.add(task);
+        storage.saveToFile(tasks);
+        System.out.println("    ____________________________________________________________");
```
Maybe you could consider having a static method like "linePrinter" in another UI class that can be called to print this line of dashes instead of manually writing them?

sp4ce-cowboy commented on src/main/java/chatbot/task/TaskManager.java:
```diff
@@ -0,0 +1,216 @@
+     * @param end end date
+     * @throws ChatbotException when one of the param is empty
+     */
+    public void addEvents(String t, String start, String end) throws ChatbotException {
```
I like how you have accounted for all possible input errors here, and your other task parsing functions as well!

sp4ce-cowboy commented on src/test/java/TaskTest.java:
```diff
@@ -0,0 +1,15 @@
+    class TaskTest {
+
+        @Test
+        void testMarkAsDone() {
```
Maybe you can try re-naming your tests according to the `unitBeingTested_descriptionOfTestInputs_expectedOutcome` convention?
```suggestion
        void taskMarker_inputTask_taskMarkedAsDone() {
```

sp4ce-cowboy reviewed (commented):
Overall your code looks fantastic, especially your extensive use of polymorphism and substitutability in close alignment with OOP Principles. Just a few minor nits here and there for you to consider, nonetheless, LGTM!

elaineshijie commented on src/main/java/chatbot/task/Task.java:
```diff
@@ -0,0 +1,116 @@
+        Task task = null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+        switch (type) {
```
Would it be better if the switch and case statements have the same indentations?

elaineshijie commented on src/main/java/chatbot/task/TaskManager.java:
```diff
@@ -0,0 +1,216 @@
+     * @param index index of the task
+     * @throws ChatbotException when invalid index is given
+     */
+    public void deleteTask(int index) throws ChatbotException {
```
I like how the methods are named with verbs and written in camelCase. Good job!

elaineshijie commented on src/main/java/chatbot/task/Events.java:
```diff
@@ -0,0 +1,40 @@
+     * constructor for class Events.
+     *
+     * @param description string for deadline's description
+     * @param startDateTime LocalDateTime for start date of the event
```
Perhaps it would be good to include the endDateTime parameter for Javadocs too?

elaineshijie commented on src/main/java/chatbot/storage/Storage.java:
```diff
@@ -0,0 +1,87 @@
+package chatbot.storage;
+
+import java.io.*;
```
Perhaps it would be better to list imported class explicitly?

elaineshijie commented on src/main/java/chatbot/task/TaskManager.java:
```diff
@@ -0,0 +1,216 @@
+        }
+    }
+
+    public void findTaskByKeyboard(String keyword) {
```
Would it be better to include Javadocs comment for this method too?

elaineshijie reviewed (commented):
Code is very concise with the use of Enums & Command class. Good job! :)

tanteckfang committed:
Merge pull request #7 from tanteckfang/branch-Level-10

done level 10

tanteckfang committed:
Add assertions

* class Chatbot.java ChatbotController.java Parser.java DialogBox.java has add some assertions in the public method such that to ensure the input or the variable is not null.
*class UI also been removed because it is replaced by the ChatbotController class and it has no usage anymore

tanteckfang committed:
Correct some of the quality for my code.

- Try to keep the use of Eygptian style.
- Try to delete some unnecessary code and method.
- Replace chatbot.task.* with the specific class used in Parser class
- Add some Javadocs

tanteckfang committed:
my find are able to find task provided partial keywords or case insensitive

tanteckfang committed:
Merge branch 'branch-A-Assertions'

tanteckfang committed:
Merge branch 'branch-A-CodeQuality'

tanteckfang committed:
Merge branch 'branch-C-BetterSearch'

tanteckfang committed:
add some credits and javadocs, delete methods with no usage

tanteckfang committed:
Resolved conflicts

tanteckfang committed:
rename chatbot'

tanteckfang committed:
add readme.md

tanteckfang committed:
correct some javastyle

tanteckfang committed:
Merge pull request #10 from tanteckfang/branch-A-UserGuide

correct some javastyle

tanteckfang committed:
fix JAR file

tanteckfang committed:
Merge pull request #11 from tanteckfang/branch-A-UserGuide

fix JAR file

tanteckfang committed:
makes delete looks better

tanteckfang committed:
Merge pull request #12 from tanteckfang/branch-A-UserGuide

makes delete looks better
